packages/public-api/src/packageInfo/packageInfoService.ts

```diff
@@ -0,0 +1,54 @@
+import { getRepository } from 'typeorm';
+import {
+  PackageMetadata,
+  PackageUsageByHostnameProjection,
+  PackagePopularityView,
+  PackageVulnerability,
+  toSerializable,
+} from '@gradejs-public/shared';
+
+export async function getPackageInfoByName(packageName: string) {
+  const packageRepo = getRepository(PackageMetadata);
+  const packageUsageRepo = getRepository(PackageUsageByHostnameProjection);
+  const packagePopularityRepo = getRepository(PackagePopularityView);
+  const packageVulnerabilities = getRepository(PackageVulnerability);
+
+  const queries = [
+    packageRepo.createQueryBuilder().where('name = :packageName', { packageName }),
+    packageUsageRepo
+      .createQueryBuilder('usage')
+      .leftJoinAndSelect('usage.hostname', 'hostname')
+      .leftJoinAndSelect('usage.sourceScan', 'sourceScan')
```

Comment:
AFAIR, there's no need for `scan` itself on the page, just the hostname, or am i missing something?

Reply:
It's here to get the count of detected packages for each hostname. I think I'll leave it as is for now, and leave a TODO comment to store that count to the projection view.